modules/apps/headless/headless-object/headless-object-test/src/testIntegration/java/com/liferay/headless/object/resource/v1_0/test/CollaboratorResourceTest.java

```diff
@@ -6,14 +6,572 @@
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.test.util.UserGroupTestUtil;
+import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.kernel.util.CalendarFactoryUtil;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.test.rule.FeatureFlags;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.vulcan.util.LocalizedMapUtil;
+import com.liferay.sharing.model.SharingEntry;
+import com.liferay.sharing.security.permission.SharingEntryAction;
+import com.liferay.sharing.service.SharingEntryLocalService;
 
-import org.junit.Ignore;
+import java.io.Serializable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
 import org.junit.runner.RunWith;
 
 /**
  * @author Alicia García
  */
-@Ignore
+@FeatureFlags("LPD-17564")
 @RunWith(Arquillian.class)
 public class CollaboratorResourceTest extends BaseCollaboratorResourceTestCase {
+
+	@BeforeClass
+	public static void setUpClass() throws Exception {
+		_objectDefinition = _getObjectDefinition();
+	}
+
+	@AfterClass
+	public static void tearDownClass() throws Exception {
+		_objectDefinitionLocalService.deleteObjectDefinition(_objectDefinition);
+	}
+
+	@Before
+	@Override
+	public void setUp() throws Exception {
+		super.setUp();
+
+		_users = new ArrayList<>();
+		_userGroups = new ArrayList<>();
+	}
+
+	@After
+	@Override
+	public void tearDown() throws Exception {
+		super.tearDown();
+
+		for (User user : _users) {
+			_userLocalService.deleteUser(user);
+		}
+
+		for (UserGroup userGroup : _userGroups) {
+			_userGroupLocalService.deleteUserGroup(userGroup);
+		}
+	}
+
+	@Override
+	@Test
+	public void testPostObjectEntryCollaboratorsPage() throws Exception {
+		ObjectEntry objectEntry = _addObjectEntry();
+
+		_addUserCollaborator(
+			_classNameLocalService.getClassNameId(
+				_objectDefinition.getClassName()),
+			objectEntry.getObjectEntryId());
+
+		Collaborator collaborator1 = _getUserCollaborator();
+		Collaborator collaborator2 = _getUserCollaborator();
+		Collaborator collaborator3 = _getUserGroupCollaborator();
+
+		Page<Collaborator> collaboratorPage =
+			collaboratorResource.postObjectEntryCollaboratorsPage(
+				objectEntry.getObjectEntryId(),
+				new Collaborator[] {
+					collaborator1, collaborator2, collaborator3
+				});
+
+		List<Collaborator> collaborators =
```

Comment:
@Mikellorza, we have a lot of duplicate code for assertContains the 3 collaborators. Could we maybe add a method that receives two arrays and check this, something like

```
private void _assertCollaborators(List<Collaborator> collaborators, Colaborator ...existingCollaborators) {
Assert.assertEquals(collaborators.toString(), existingCollaborators.size(), collaborators.size());

for(Collaborator existingCollaborator : existingCollaborators) {
assertContains(existingCollaborator, collaborators);
}
}
```

sorry, wrote it from memory and without intellij :D

Reply:
Yes, I am going to improve it, I think it is better to have that method and the code it will be better.

thanks mozo